index.js

```diff
@@ -248,22 +248,43 @@ module.exports = function (config) {
         }, function () {
           var marker = markers[markerIndex];
           var p;
+
           markerIndex++;
           if (marker.type === 'Capture') {
             p = timeHandler.goToTimeAndAnimateForCapture(browserFrames, marker.time);
             // because this section is run often and there is a small performance
             // penalty of using .then(), we'll limit the use of .then()
             // to only if there's something to do
-            if (config.preparePageForScreenshot) {
+
+            if (config.skipScreenshot) {
               p = p.then(function () {
+                return config.skipScreenshot(page, marker.data.frameCount, framesToCapture);
+              })
+            }
```

Comment:
I think `config.shouldSkipFrame` is a better name for the function. For many argument functions, I'm going to be shifting towards object literals (e.g. `fn({ a: true, b: false ...})`), to make it easier to add more arguments in the future, so this new function should use that convention.

I think passing a variable down a promise chain is clunky, I'd prefer if it were just stored to a local variable.

```suggestion
            var skipCurrentFrame;
            if (config.shouldSkipFrame) {
              p = p.then(function () {
                skipCurrentFrame = config.shouldSkipFrame({
                  page: page,
                  frameCount: marker.data.frameCount,
                  framesToCapture: framesToCapture
                });
              })
            }
```